### PR TITLE
relax label name comparison in benchmarks.yaml

### LINKED ISSUE #6518: "run-benchmarks" label is triggering workflow which is then skipped
## Description

Adding the "run-benchmarks" label to a PR is supposed to trigger our [benchmark workflow](https://github.com/scikit-image/scikit-image/blob/main/.github/workflows/benchmarks.yml#L13). Adding this seems to trigger "something" and causes to GitHub to display that workflow. However, it's always shown as "Skipped".

E.g. https://github.com/scikit-image/scikit-image/pull/3255/commits/be1eb8c168f60b1987700ea9d219d00dd7d12878, https://github.com/scikit-image/scikit-image/pull/6509#issuecomment-1242954025 (https://github.com/scikit-image/scikit-image/pull/6509/commits/1d0e9d377f1836f911d2ee04271432a6dbf2a7ec), https://github.com/scikit-image/scikit-image/pull/6511/commits/70322e6dcc4b9cf89a13fe5e115393a77202d65a, ...

I think https://github.com/scikit-image/scikit-image/pull/6182/commits/b589052bb8884e07249f2eefbde3e8ee07f3dfc5 is the most recent commit I could find where it still works.

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #6518

~This change from https://github.com/scikit-image/scikit-image/pull/6426 seems to prevent running the benchmarks via the run-benchmark label in PRs.~

~Will try adding the label here to verify...~

**update**: The problem was not permissions, but a change which included an emoji in the benchmark tag. The string comparison is relaxed in this PR to allow the workflow to run in this case as well.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Comment by grlee77** [comment]
Well, the label still did not work when I tried it, but I am not certain if this would have to be merged first for it to take effect?

**Review by lagru — COMMENTED** [review]
Could you bump the Python version of `actions/setup-python@v3` to 3.10 as well? We updated asv's config in https://github.com/scikit-image/scikit-image/commit/70322e6dcc4b9cf89a13fe5e115393a77202d65a so it fails with the currently used Python 3.8: https://github.com/scikit-image/scikit-image/actions/runs/3068608577/jobs/4956277449#step:7:79

**Comment by grlee77** [comment]
Okay, the issue turned out not to be permissions-related but was instead because at some point a "bar-chart" emoji was added to the label and which caused the exact label name match to fail.

I changed here to a much more lenient match for any label string containing "benchmark"

**Comment by grlee77** [comment]
Before merging we should clean up all the irrelevant debugging commit messages and squash this to just a single commit with text something like:
```
* relax exact label name comparison - match any label with "benchmark" in the name
* update to use Python 3.10 as in `asv.conf.json`
```

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ permissions:
 
 jobs:
   benchmark:
-    if: ${{ github.event.label.name == 'run-benchmark' && github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' }}
+    if: contains(github.event.label.name, 'benchmark') || github.event_name == 'workflow_dispatch'
```
Comment (lagru) [inline .github/workflows/benchmarks.yml:13]: For those wondering like I did, [`contains` is a GitHub specific expression](https://docs.github.com/en/actions/learn-github-actions/expressions#contains).

**Review by lagru — APPROVED** [review]
Great catch @grlee77! I should have realized because I had a similar thought process [while configuring the stale action](https://github.com/scikit-image/scikit-image/pull/6506/files#diff-2e6f9d0d5bc3a93c35d2ea54e7ac119b0d658292ceb90f0a7a1283c79e7fc0ccR20). :sweat_smile:

File: .github/workflows/benchmarks.yml (hunk 1 above)
Comment (grlee77) [reply .github/workflows/benchmarks.yml:13]: yeah, I had to do a search to see if there was a comparison function and came across that page :)

**Pull request merged**